Marlin/src/libs/nozzle.cpp

```diff
@@ -159,22 +159,49 @@ Nozzle nozzle;
   void Nozzle::clean(const uint8_t &pattern, const uint8_t &strokes, const float &radius, const uint8_t &objects, const uint8_t cleans) {
     xyz_pos_t start[HOTENDS] = NOZZLE_CLEAN_START_POINT, end[HOTENDS] = NOZZLE_CLEAN_END_POINT, middle[HOTENDS] = NOZZLE_CLEAN_CIRCLE_MIDDLE;
 
+    const uint8_t arrPos = ANY(SINGLENOZZLE, MIXING_EXTRUDER) ? 0 : active_extruder;
+
+    #if HAS_SOFTWARE_ENDSTOPS
+
+      #define LIMIT_AXIS(A) do{ \
+        LIMIT( start[arrPos].A, soft_endstop.min.A, soft_endstop.max.A); \
+        LIMIT(middle[arrPos].A, soft_endstop.min.A, soft_endstop.max.A); \
+        LIMIT(   end[arrPos].A, soft_endstop.min.A, soft_endstop.max.A); \
+      }while(0)
+
+      LIMIT_AXIS(x);
+      LIMIT_AXIS(y);
+      LIMIT_AXIS(z);
+
+      const float arc_part = radius / M_PI;
```

Comment:
The radius divided by 3.14 is interesting…. It looks like the radius itself is what you probably want here instead. The circles will extend from the middle of some point out to the radius distance.

Reply:
@InsanityAutomation — I can't merge this without some kind of answer.

Reply:
I'm going to assume I am correct and replace `arc_part` below with `radius`.

Reply:
Sorry for the delay, it's been very crazy around here lately dealing with some stuff. Yes you are correct.